Improve compose box behavior when user does not have permission to post in a stream.

sahil839 opened the pull request:
<!-- Describe your pull request here.-->
- First commit is to show all accessible streams, including the ones to which user cannot post, in the dropdown.
- Second commit is to show the banner on opening compose box or changing the stream, if user is not allowed to post.

Fixes: <!-- Issue link, or clear description.--> #25219 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![post-banner](https://user-images.githubusercontent.com/35494118/233839898-6f3f8888-b505-4f24-92eb-2ba12bfb960f.png)
![post-banner](https://user-images.githubusercontent.com/35494118/233839908-de405651-2352-4cd6-82f0-592c08209416.gif)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

sahil839 commented:
@alya can you test this?
Also do we want to show all the streams in the dropdown menu in "Move message" option as well. Not sure where we would show the banner in that case.

alya commented:
Hm, the banner looks good, but I'm somehow getting stuck in a "can't post" state.

![Screen Shot 2023-04-24 at 1 07 44 PM](https://user-images.githubusercontent.com/2090066/234104758-fa0eafba-c45c-4f18-99f6-991f60b24df6.png)

alya commented:
Can we change the color of the "Send" button and `^` next to it to make it visually clear that it's disabled?

alya commented:
> Also do we want to show all the streams in the dropdown menu in "Move message" option as well. Not sure where we would show the banner in that case.

Good question! Yeah, I think showing all the streams + a banner is the right approach in "Move messages" / "Move topic" as well. I guess the banner would go between the modal title and body.

This part can be done as a follow-up PR, and doesn't have to be completed for the release, since it's much more of an edge case.

sahil839 commented:
Updated the PR -
- Fixed the bug of being stuck in "can't post" state.
- Changed the color of button to a shade of grey to show it is disabled.
![stream-post-compose](https://user-images.githubusercontent.com/35494118/234358753-4f507014-0bd1-42bb-86af-c7e16c25de0d.png)

alya commented:
Hmm, the disabled "Send" button color seems off to me. I definitely think it doesn't look good in dark theme:

![Screen Shot 2023-04-25 at 12 32 35 PM](https://user-images.githubusercontent.com/2090066/234383530-90dd932a-f108-42c1-ab8b-b1c71f936f06.png)

vs. 
![Screen Shot 2023-04-25 at 12 32 23 PM](https://user-images.githubusercontent.com/2090066/234383547-2053d05f-9318-44ea-a3f9-cd6df9884c44.png)

It also looks too light to me in light theme.

sahil839 commented:
Updated the background color of buttons in disabled state to be same as that in case of modals. Also updated the newly added class to be more clear and use `-`s instead of `_`s.

![send-disabled](https://user-images.githubusercontent.com/35494118/234462875-d96dd64d-1cde-4dc9-9f02-22089f1c1d95.png)
![send-disabled-dark](https://user-images.githubusercontent.com/35494118/234462884-36269187-30fe-4ee1-8af9-91ae0bf3826b.png)

sahil839 commented on web/styles/dark_theme.css:
```diff
@@ -59,11 +59,6 @@
-    .modal__btn:disabled {
-        opacity: 1;
-        background-color: hsl(0deg 0% 83% / 50%);
-    }
```
Moved this below since we want same CSS for compose buttons in disabled state as well and here we only define CSS for modals.

alya commented:
Thanks, those screenshots look good to me!

timabbott commented:
Merged, thanks @sahil839!
